src/test/java/uk/gov/register/util/HashValueTest.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.register.util;
+
+import org.junit.Test;
+import uk.gov.register.core.HashingAlgorithm;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class HashValueTest {
```

Comment:
Can we add tests for equals and hashcode for when the `hashingAlgorithm` is different?

Also can we add tests for cases where decode will fail and the exceptions we expect?